src/components/OSCALCatalogGroup.test.js

```diff
@@ -2,37 +2,53 @@ import React from "react";
 import { render, screen, fireEvent } from "@testing-library/react";
 import OSCALCatalogGroup from "./OSCALCatalogGroup";
 
+const testGroup = {
+  id: "parent-group",
+  class: "family",
+  title: "Parent Group",
+  groups: [
+    {
+      id: "ac",
+      class: "family",
+      title: "Access Control",
+      controls: [{ id: "ac-1" }],
+    },
+  ],
+};
+
 test("OSCALCatalog displays control groups", () => {
-  const testGroup = {
-    id: "ac",
-    class: "family",
-    title: "Access Control",
-    controls: [{ id: "ac-1" }],
-  };
   render(<OSCALCatalogGroup group={testGroup} />);
+
+  const parentGroup = screen.getByText("Parent Group");
+  fireEvent.click(parentGroup);
   const result = screen.getByText("Access Control");
   expect(result).toBeVisible();
 });
 
 test("OSCALCatalog displays nested control groups", () => {
-  const testGroup = {
-    id: "parent-group",
-    class: "family",
-    title: "Parent Group",
-    groups: [
-      {
-        id: "ac",
-        class: "family",
-        title: "Access Control",
-        controls: [{ id: "ac-1" }],
-      },
-    ],
-  };
   render(<OSCALCatalogGroup group={testGroup} />);
+
   const parentGroup = screen.getByText("Parent Group");
   fireEvent.click(parentGroup);
   const childGroup = screen.getByText("Access Control");
   fireEvent.click(childGroup);
   const control = screen.getByText("ac-1");
   expect(control).toBeVisible();
 });
+
+test("OSCALCatalogGroup displays legend", async () => {
+  render(<OSCALCatalogGroup group={testGroup} />);
+
+  const parentGroup = screen.getByText("Parent Group");
+  fireEvent.click(parentGroup);
+
+  const placeholderBox = screen.getByLabelText("legend-placeholder-label");
+  expect(placeholderBox).toBeVisible();
+  const placeholderBoxLabel = screen.getByText("Placeholder");
+  expect(placeholderBoxLabel).toBeVisible();
+
+  const valueBox = screen.getByLabelText("legend-value-label");
+  expect(valueBox).toBeVisible();
+  const valueBoxLabel = screen.getByText("Value");
+  expect(valueBoxLabel).toBeVisible();
```

Comment:
This seems very similar to:

https://github.com/EasyDynamics/oscal-react-library/blob/d002755818b29fbc4f357b441836d08741675430/src/components/OSCALComponentDefinitionControlImplementation.test.js#L60-L68

If it is easy to reduce this repetition, it may be worth looking into. I would not waste much time doing it however.

Reply:
This check is being performed on each of the control groupings for the different models. This could be wrapped into a function pulled from one of these test suites or a new test file could be created specifically for these tests. Any suggestions?

Reply:
`ControlParamLegend.test.js` will solve this. This helper test class is used for each of the control groups to reference, and avoids code duplication. In order for this to be a test class, I needed to include a test and supplied a basic `OSCALCatalogGroup` test.